shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.mm

```diff
@@ -0,0 +1,318 @@
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.h"
+
+#import "flutter/shell/platform/embedder/embedder.h"
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.h"
+
+namespace flutter {  // namespace
+
+bool SystemVersionEqualTo(NSOperatingSystemVersion target) {
+  NSOperatingSystemVersion systemVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
+  return target.majorVersion == systemVersion.majorVersion &&
+         target.minorVersion == systemVersion.minorVersion;
+}
+
+bool SystemVersionGreaterOrEqualTo(NSOperatingSystemVersion target) {
+  NSOperatingSystemVersion systemVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
+  if (systemVersion.majorVersion < target.majorVersion) {
+    return false;
+  } else if (systemVersion.majorVersion > target.majorVersion) {
+    return true;
+  }
+  return systemVersion.minorVersion >= target.minorVersion;
+}
+
+bool SystemVersionLessOrEqualTo(NSOperatingSystemVersion target) {
+  NSOperatingSystemVersion systemVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
+  if (systemVersion.majorVersion < target.majorVersion) {
+    return true;
+  } else if (systemVersion.majorVersion > target.majorVersion) {
+    return false;
+  }
+  return systemVersion.minorVersion <= target.minorVersion;
+}
+
+// Native mac notifications fired.
+NSString* const NSAccessibilityLoadCompleteNotification = @"AXLoadComplete";
+NSString* const NSAccessibilityInvalidStatusChangedNotification = @"AXInvalidStatusChanged";
+NSString* const NSAccessibilityLiveRegionCreatedNotification = @"AXLiveRegionCreated";
+NSString* const NSAccessibilityLiveRegionChangedNotification = @"AXLiveRegionChanged";
+NSString* const NSAccessibilityExpandedChanged = @"AXExpandedChanged";
+NSString* const NSAccessibilityMenuItemSelectedNotification = @"AXMenuItemSelected";
+
+AccessibilityBridgeMacDelegate::AccessibilityBridgeMacDelegate(FlutterEngine* flutterEngine)
+    : _flutterEngine(flutterEngine) {}
+
+void AccessibilityBridgeMacDelegate::OnAccessibilityEvent(
+    ui::AXEventGenerator::TargetedEvent targeted_event) {
+  ui::AXNode* ax_node = targeted_event.node;
+  ui::AXEventGenerator::Event event_type = targeted_event.event_params.event;
+  auto bridge = _flutterEngine.accessibilityBridge.lock();
+  NSCAssert(bridge, @"Accessibility bridge in flutter engine must not be null.");
+  auto platform_node_delegate = bridge->GetFlutterPlatformNodeDelegateFromID(ax_node->id()).lock();
+  NSCAssert(platform_node_delegate, @"Event target must exist in accessibility bridge.");
+
+  auto mac_platform_node_delegate =
+      std::static_pointer_cast<FlutterPlatformNodeDelegateMac>(platform_node_delegate);
+  gfx::NativeViewAccessible native_node = mac_platform_node_delegate->GetNativeViewAccessible();
+
+  NSString* mac_notification = nil;
+
+  switch (event_type) {
```

Comment:
I think I'd find this a bit easier to read through if it was moved to its own method, and returned a `std::optional<NSString>` or something like that.

Alternatively, if it returned some kind of object that has a String and native_node (and whatever else might be needed) so that the cases are handled more consistently with just a `return` and we're not thinking about which ones fall through or break or return.

Right now there are a few different paths this can flow through and it might be less confusing for future us when we go to handle new cases or change

Reply:
this is now a separate function that will return a list of NSAccessibilityEvent

Reply:
This is definitely an improvement; I'd be tempted to implement this as taking ax_node and returning a single `NSAccessibilityEvent` rather than a list of them since it's more straightforward to test and increases the utility if it should ever be needed in another context.

Reply:
I am not sure if i understand. It is possible that it may generate more than one event for an input AXEvent. Can you explain more?